nixos/lib/eval-config.nix

```diff
@@ -57,14 +53,12 @@ let
         # they way through, but has the last priority behind everything else.
         nixpkgs.system = lib.mkDefault system;
       })
-      ++
-      (optional (pkgs_ != null) {
-        _module.args.pkgs = lib.mkForce pkgs_;
-      })
```

Comment:
We shouldn't change the behavior with just a warning, and I feel like this is too big a change anyway - as this is the stable NixOS entrypoint.

How about we start with removing `pkgs` from `lib.nixosSystem`? That's a weird one, where it's unclear what the intent should even be: ignore `nixpkgs.*`? Use `readOnlyPkgs`? I think both are equally surprising and users should make a choice (if they pass `pkgs` at all).

Reply:
My intent is making old guides which use a pattern that looks like this:

```nix
  outputs = { nixpkgs, ... }:
  let
    system = "x86_64-linux";
    pkgs = import nixpkgs {
      inherit system;
      config.allowUnfree = true;
   };
  in {
    nixosConfigurations.hostname = nixpkgs.lib.nixosSystem {
      inherit system;
      inherit pkgs; #this
      specialArgs = { inherit pkgs; };#or this
      };
    }
```
Stop shooting people in the foot